### ISSUE TITLE
Fatal error: Can't remove more items from a collection than it contains

### ISSUE THREAD

**Issue by b95505017** [arkivanov/Decompose#298]
I just clone this project and ran sample on iOS 16.2 simulator. When I press `Next` and pop back got this crash:

```
Swift/RangeReplaceableCollection.swift:829: Fatal error: Can't remove more items from a collection than it contains
2023-01-08 05:13:46.297584+0800 app-ios[5447:74543] Swift/RangeReplaceableCollection.swift:829: Fatal error: Can't remove more items from a collection than it contains
```

**Comment by arkivanov** [comment]
Thanks for the report. Could you please provide a full stack trace?

**Comment by b95505017** [comment]
@arkivanov Sorry I only see these two lines in my XCode. If there are some ways to receive more log, I'd like to help.
<img width="1512" alt="截圖 2023-01-08 下午5 56 25" src="https://user-images.githubusercontent.com/320815/211190009-295de60c-4c36-4a17-9295-3c70d01eae25.png">

**Comment by b95505017** [comment]
@arkivanov  I've attached the screenshot to see if it could help you debug. Also it won't crash if I switch to `StackInteropView` instead of `NavigationStack`.

**Comment by arkivanov** [comment]
Thanks, I couldn't reproduce the error so far. Could you please describe what exact buttons you click? A video would help as well.

**Comment by arkivanov** [comment]
Ok, I was able to reproduce the issue. For me it happens pretty rarely when navigating back, not every time. I will check this.

**Comment by arkivanov** [comment]
Here is the stack trace just in case:

```
* thread #1, queue = 'com.apple.main-thread', stop reason = Fatal error: Can't remove more items from a collection than it contains
    frame #0: 0x000000018bfa78e0 libswiftCore.dylib`_swift_runtime_on_report
    frame #1: 0x000000018c034580 libswiftCore.dylib`_swift_stdlib_reportFatalErrorInFile + 204
    frame #2: 0x000000018bce2e10 libswiftCore.dylib`closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in Swift._assertionFailure(_: Swift.StaticString, _: Swift.StaticString, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 356
    frame #3: 0x000000018bce2b78 libswiftCore.dylib`closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in Swift._assertionFailure(_: Swift.StaticString, _: Swift.StaticString, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 192
    frame #4: 0x000000018bce2564 libswiftCore.dylib`Swift._assertionFailure(_: Swift.StaticString, _: Swift.StaticString, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 228
    frame #5: 0x000000018bcdc864 libswiftCore.dylib`Swift.RangeReplaceableCollection< where τ_0_0: Swift.BidirectionalCollection, τ_0_0 == τ_0_0.Swift.Collection.SubSequence>._customRemoveLast(Swift.Int) -> Swift.Bool + 1332
    frame #6: 0x000000018bcdc0cc libswiftCore.dylib`protocol witness for Swift.RangeReplaceableCollection._customRemoveLast(Swift.Int) -> Swift.Bool in conformance Swift.ArraySlice<τ_0_0> : Swift.RangeReplaceableCollection in Swift + 52
    frame #7: 0x000000018be0cff0 libswiftCore.dylib`merged Swift.RangeReplaceableCollection< where τ_0_0: Swift.BidirectionalCollection>.removeLast(Swift.Int) -> () + 460
    frame #8: 0x000000010543436c SwiftUI`___lldb_unnamed_symbol174259 + 76
    frame #9: 0x00000001051c91d8 SwiftUI`___lldb_unnamed_symbol152317 + 3136
    frame #10: 0x00000001051c60bc SwiftUI`___lldb_unnamed_symbol152311 + 2300
    frame #11: 0x000000010564bd54 SwiftUI`___lldb_unnamed_symbol188680 + 3184
    frame #12: 0x00000001052276ec SwiftUI`___lldb_unnamed_symbol155160 + 408
    frame #13: 0x0000000105228c50 SwiftUI`___lldb_unnamed_symbol155181 + 20
    frame #14: 0x000000010529c1b0 SwiftUI`___lldb_unnamed_symbol159348 + 20
    frame #15: 0x00000001050053c0 SwiftUI`___lldb_unnamed_symbol139732 + 464
    frame #16: 0x0000000105004e78 SwiftUI`___lldb_unnamed_symbol139731 + 80
    frame #17: 0x00000001052274b4 SwiftUI`___lldb_unnamed_symbol155158 + 588
    frame #18: 0x00000001055548f8 SwiftUI`___lldb_unnamed_symbol180833 + 1968
    frame #19: 0x00000001055549a0 SwiftUI`___lldb_unnamed_symbol180834 + 68
    frame #20: 0x0000000101a9b7f8 UIKitCore`-[UINavigationController navigationTransitionView:didEndTransition:fromView:toView:] + 1428
    frame #21: 0x0000000101a92f40 UIKitCore`__49-[UINavigationController _startCustomTransition:]_block_invoke + 204
    frame #22: 0x0000000101b5a55c UIKitCore`-[_UIViewControllerTransitionContext completeTransition:] + 104
    frame #23: 0x0000000101b69a10 UIKitCore`__53-[_UINavigationParallaxTransition animateTransition:]_block_invoke_5 + 580
    frame #24: 0x000000010264fee0 UIKitCore`__UIVIEW_IS_EXECUTING_ANIMATION_COMPLETION_BLOCK__ + 28
    frame #25: 0x0000000102650144 UIKitCore`-[UIViewAnimationBlockDelegate _didEndBlockAnimation:finished:context:] + 604
    frame #26: 0x000000010262b7ac UIKitCore`-[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 212
    frame #27: 0x000000010262bbe8 UIKitCore`-[UIViewAnimationState animationDidStop:finished:] + 192
    frame #28: 0x000000010262bce8 UIKitCore`-[UIViewAnimationState animationDidStop:finished:] + 448
    frame #29: 0x0000000187f243a4 QuartzCore`CA::Layer::run_animation_callbacks(void*) + 228
    frame #30: 0x0000000100af1d5c libdispatch.dylib`_dispatch_client_callout + 16
    frame #31: 0x0000000100b02808 libdispatch.dylib`_dispatch_main_queue_drain + 1316
    frame #32: 0x0000000100b022d4 libdispatch.dylib`_dispatch_main_queue_callback_4CF + 40
    frame #33: 0x0000000180372cf4 CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
    frame #34: 0x000000018036d3b0 CoreFoundation`__CFRunLoopRun + 1956
    frame #35: 0x000000018036c7f4 CoreFoundation`CFRunLoopRunSpecific + 584
    frame #36: 0x0000000188faec98 GraphicsServices`GSEventRunModal + 160
    frame #37: 0x00000001021c05d4 UIKitCore`-[UIApplication _run] + 868
    frame #38: 0x00000001021c45cc UIKitCore`UIApplicationMain + 124
    frame #39: 0x0000000105816b18 SwiftUI`___lldb_unnamed_symbol203530 + 164
    frame #40: 0x00000001058169c0 SwiftUI`___lldb_unnamed_symbol203528 + 140
    frame #41: 0x0000000104fba36c SwiftUI`static SwiftUI.App.main() -> () + 80
  * frame #42: 0x0000000100409fa4 app-ios`static app_iosApp.$main(self=app_ios.app_iosApp) at app_iosApp.swift:11:1
    frame #43: 0x000000010040adec app-ios`main at app_iosApp.swift:0
    frame #44: 0x000000010066dfa0 dyld_sim`start_sim + 20
    frame #45: 0x0000000100771e50 dyld`start + 2544
```

**Comment by arkivanov** [comment]
For now it looks like a bug in SwiftUI.

**Comment by b95505017** [comment]
https://user-images.githubusercontent.com/320815/211208478-4c5c644a-5bfe-4142-8e93-9cbbec729db8.mov

Video attached. Crash everytime.
Did you mean that this issue could be reproducible in pure SwiftUI?

**Comment by arkivanov** [comment]
At the moment I believe it should be reproducible in pure SwiftUI. I will play with it and provide more updates here.

**Comment by arkivanov** [comment]
If looks like I managed to fix the crash - #299. Not sure whether it's a bug in SwiftUI or an incorrect usage from my side.

**Comment by b95505017** [comment]
LGTM 👍
